packages/itwin/ec3-widget/src/components/ExportModal.tsx

```diff
@@ -0,0 +1,206 @@
+        if (job.id && token) {
+          const currentJobStatus =
+            await ec3JobsClient.getEC3JobStatus(token, job.id);
+          if (currentJobStatus.status) {
+            if (
+              currentJobStatus.status === CarbonUploadState.Succeeded
+            ) {
+              setJobLink(currentJobStatus._links.ec3Project);
+            }
+            setJobStatus(currentJobStatus.status);
+          } else {
+            setJobStatus(CarbonUploadState.Failed);
+            toaster.negative("Failed to get job status. 😔");
+          }
+        }
+      }, PIN_INTERVAL);
+      intervalRef.current = intervalId;
+    },
+    [setJobLink, setJobStatus, ec3JobsClient]
+  );
+
+  const runJob = useCallback(
+    async (token: string) => {
+      const accessToken =
+        (await IModelApp.authorizationClient?.getAccessToken()) ?? "";
+      if (props.templateId && token) {
+        try {
+          const jobRequest: EC3JobCreate = {
+            configurationId: props.templateId,
+            projectName: props.projectName,
+            ec3BearerToken: token,
+          }
+          const jobCreated = await ec3JobsClient.createJob(
+            accessToken,
+            jobRequest
+          );
+          if (jobCreated.id) {
+            pinStatus(jobCreated);
+          } else {
+            setJobStatus(CarbonUploadState.Failed);
+            toaster.negative("Failed to create EC3 job. 😔");
+          }
+        } catch (e) {
+          setJobStatus(CarbonUploadState.Failed);
+          toaster.negative("You do not have the required permissions. Please contact the project administrator.");
+          /* eslint-disable no-console */
+          console.error(e);
+        }
+      } else {
+        setJobStatus(CarbonUploadState.Failed);
+        toaster.negative("Invalid reportId.");
+      }
+    },
+    [props, pinStatus, ec3JobsClient]
+  );
+
+  const onClose = useCallback(() => {
+    setJobStatus(undefined);
+    setJobLink(undefined);
+    window.clearInterval(intervalRef.current);
+    props.close();
+  }, [props]);
+
+  const getStatusComponent = useCallback(
+    (status: CarbonUploadState, link: string | undefined) => {
+      switch (status) {
+        case CarbonUploadState.Queued:
+          return (
+            <div className="ec3w-progress-radial-container">
+              <ProgressRadial indeterminate size="small" value={50} />
+              <Text variant="leading" className="ec3w-status-text">
+                Export queued
+              </Text>
+            </div>
+          );
+        case CarbonUploadState.Running:
+          return (
+            <div className="ec3w-progress-linear-container">
+              <ProgressLinear indeterminate />
+              <Text variant="leading" className="ec3w-status-text">
+                Export running
+              </Text>
+            </div>
+          );
+        case CarbonUploadState.Succeeded:
+          return (
+            link && (
```

Comment:
What happens if `link` is falsy?

Reply:
Applied the suggested change from powerBI